src/qibolab/instruments/rfsoc.py

```diff
@@ -0,0 +1,501 @@
+                # convert results
+                res = self.convert_sweep_results(sweepers[0], original_ro, sequence, qubits, toti, totq, average)
+                return res
+
+            # if it's not possible to execute qick sweep re-call function
+            else:
+                sweep_results = {}
+                idx_pulse = or_sequence.index(sweeper.pulses[0])
+                for val in sweeper.values:
+                    if is_freq:
+                        sequence[idx_pulse].frequency = val
+                    elif is_amp:
+                        sequence[idx_pulse].amplitude = val
+                    res = self.recursive_python_sweep(qubits, sequence, or_sequence, *sweepers[1:], average=average)
+                    # merge the dictionary obtained with the one already saved
+                    sweep_results = self.merge_sweep_results(sweep_results, res)
+
+        return sweep_results
+
+    def merge_sweep_results(
+        self,
+        dict_a: Dict[str, Union[AveragedResults, ExecutionResults]],
+        dict_b: Dict[str, Union[AveragedResults, ExecutionResults]],
+    ) -> Dict[str, Union[AveragedResults, ExecutionResults]]:
+        """Merge two dictionary mapping pulse serial to Results object.
+        If dict_b has a key (serial) that dict_a does not have, simply add it,
+        otherwise sum the two results (`qibolab.result.ExecutionResults`
+        or `qibolab.result.AveragedResults`)
+        Args:
+            dict_a (dict): dict mapping ro pulses serial to qibolab res objects
+            dict_b (dict): dict mapping ro pulses serial to qibolab res objects
+        Returns:
+            A dict mapping the readout pulses serial to qibolab results objects
+        """
+        for serial in dict_b:
+            if serial in dict_a:
+                dict_a[serial] = dict_a[serial] + dict_b[serial]
+            else:
+                dict_a[serial] = dict_b[serial]
+        return dict_a
+
+    def get_if_python_sweep(self, sequence: PulseSequence, qubits: List[Qubit], *sweepers: Sweeper) -> bool:
+        """Check if a sweeper must be run with python loop or on hardware.
+        To be run on qick internal loop a sweep must:
+            * not be on the readout frequency
+            * be just one sweeper
+            * only one pulse per channel supported (for now)
+        Args:
+            sequence (`qibolab.pulses.PulseSequence`). Pulse sequence to play.
+            qubits (list): List of `qibolab.platforms.utils.Qubit` objects
+                           passed from the platform.
+            *sweepers (`qibolab.Sweeper`): Sweeper objects.
+        Returns:
+            A boolean value true if the sweeper must be executed by python
+            loop, false otherwise
+        """
+
+        # is_amp = sweepers[0].parameter == Parameter.amplitude
+        is_freq = sweepers[0].parameter == Parameter.frequency
+
+        # if there isn't only a sweeper do a python sweep
+        if len(sweepers) != 1:
```

Comment:
Yes this is what I'm doing: the sweep_recursive function execute a single sweep at a time and if it's supported by qick does a firmware sweep.
For sure this could be extended since qick also supports multiple firmware sweep